persistence/mongo/src/main/scala/za/co/absa/spline/persistence/mongo/dao/BaselineLineageDAO.scala

```diff
@@ -235,13 +236,25 @@ abstract class BaselineLineageDAO extends VersionedLineageDAO with Logging {
     )
     val optionalTextSearchCriterion = maybeText map {
       text =>
-        val regexMatchOnFieldsCriteria = Seq("appId", "appName", "rootOperation.path") map (_ $regex quote(text) $options "i")
+        val regexMatchOnFieldsCriteria = Seq("appId", "appName", "rootOperation.path") map (_ $regex quoteSafely(text) $options "i")
         val optDatasetIdMatchCriterion = UUIDExtractor unapply text.toLowerCase map (uuid => DBObject("rootDataset._id" → uuid))
         $or(regexMatchOnFieldsCriteria ++ optDatasetIdMatchCriterion)
     }
     $and(paginationDeduplicationCriteria ++ optionalTextSearchCriterion)
   }
 
+  /**
+    * Escape special characters in a regular expression.
+    * Does not use Pattern.quote() for Cosmos DB compatibility.
+    * @see <a href="https://github.com/AbsaOSS/spline/issues/166">issue #166</a>
+    *
+    * @param text A string
+    * @return A regular expression searching for the exact literal text string
+    */
+  protected def quoteSafely(text: String): String = {
```

Comment:
Is there any reason for the method being `protected` instead of `private` ?

Reply:
overengineering probably :) thought it could be an extension point for other storage implementations